x-pack/plugins/osquery/public/agents/agents_table.tsx

```diff
@@ -5,158 +5,225 @@
  * 2.0.
  */
 
-import { find } from 'lodash/fp';
-import React, { useCallback, useEffect, useMemo, useState, useRef } from 'react';
-import {
-  EuiBasicTable,
-  EuiBasicTableColumn,
-  EuiBasicTableProps,
-  EuiTableSelectionType,
-  EuiHealth,
-} from '@elastic/eui';
+import React, { useCallback, useEffect, useState } from 'react';
+import { EuiComboBox, EuiComboBoxOptionOption, EuiHealth, EuiHighlight } from '@elastic/eui';
 
 import { useAllAgents } from './use_all_agents';
+import { useAgentGroups } from './use_agent_groups';
 import { Direction } from '../../common/search_strategy';
 import { Agent } from '../../common/shared_imports';
+import {
+  getNumAgentsInGrouping,
+  generateAgentCheck,
+  getNumOverlapped,
+  generateColorPicker,
+} from './helpers';
+
+import { AGENT_GROUP_KEY, SelectedGroups, AgentOptionValue, GroupOptionValue } from './types';
+
+export interface AgentsSelection {
+  agents: string[];
+  allAgentsSelected: boolean;
+  platformsSelected: string[];
+  policiesSelected: string[];
+}
 
 interface AgentsTableProps {
-  selectedAgents: string[];
-  onChange: (payload: string[]) => void;
+  agentSelection: AgentsSelection;
+  onChange: (payload: AgentsSelection) => void;
 }
 
-const AgentsTableComponent: React.FC<AgentsTableProps> = ({ selectedAgents, onChange }) => {
-  const [pageIndex, setPageIndex] = useState(0);
-  const [pageSize, setPageSize] = useState(5);
-  const [sortField, setSortField] = useState<keyof Agent>('upgraded_at');
-  const [sortDirection, setSortDirection] = useState<Direction>(Direction.asc);
-  const [selectedItems, setSelectedItems] = useState([]);
-  const tableRef = useRef<EuiBasicTable<Agent>>(null);
-
-  const onTableChange: EuiBasicTableProps<Agent>['onChange'] = useCallback(
-    ({ page = {}, sort = {} }) => {
-      const { index: newPageIndex, size: newPageSize } = page;
-
-      const { field: newSortField, direction: newSortDirection } = sort;
-
-      setPageIndex(newPageIndex);
-      setPageSize(newPageSize);
-      setSortField(newSortField);
-      setSortDirection(newSortDirection);
-    },
-    []
-  );
+type GroupOption = EuiComboBoxOptionOption<AgentOptionValue | GroupOptionValue>;
 
-  const onSelectionChange: EuiTableSelectionType<{}>['onSelectionChange'] = useCallback(
-    (newSelectedItems) => {
-      setSelectedItems(newSelectedItems);
+const getColor = generateColorPicker();
 
-      if (onChange) {
-        // @ts-expect-error update types
-        onChange(newSelectedItems.map((item) => item._id));
-      }
-    },
-    [onChange]
-  );
+const AgentsTableComponent: React.FC<AgentsTableProps> = ({ onChange }) => {
+  // handle paged fetching of agents
+  const [pageIndex /* , setPageIndex*/] = useState(0);
```

Comment:
are you going to fix that in the followup PR?

Reply:
yeah, i meant for the comment to be a TODO.